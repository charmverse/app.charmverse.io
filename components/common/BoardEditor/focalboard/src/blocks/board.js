"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
exports.__esModule = true;
exports.createBoard = void 0;
var utils_1 = require("../utils");
var block_1 = require("./block");
function createBoard(_a) {
    var _b;
    var _c = _a === void 0 ? {} : _a, block = _c.block, addDefaultProperty = _c.addDefaultProperty;
    addDefaultProperty = addDefaultProperty !== null && addDefaultProperty !== void 0 ? addDefaultProperty : false;
    var cardProperties = (_b = block === null || block === void 0 ? void 0 : block.fields.cardProperties.map(function (o) {
        return {
            id: o.id,
            name: o.name,
            type: o.type,
            options: o.options ? o.options.map(function (option) { return (__assign({}, option)); }) : []
        };
    })) !== null && _b !== void 0 ? _b : [];
    var selectProperties = cardProperties.find(function (o) { return o.type === 'select'; });
    if (!selectProperties && addDefaultProperty) {
        var property = {
            id: utils_1.Utils.createGuid(utils_1.IDType.BlockID),
            name: 'Status',
            type: 'select',
            options: [{
                    color: 'propColorTeal',
                    id: utils_1.Utils.createGuid(utils_1.IDType.BlockID),
                    value: 'Completed'
                }, {
                    color: 'propColorYellow',
                    id: utils_1.Utils.createGuid(utils_1.IDType.BlockID),
                    value: 'In progress'
                }, {
                    color: 'propColorRed',
                    id: utils_1.Utils.createGuid(utils_1.IDType.BlockID),
                    value: 'Not started'
                }]
        };
        cardProperties.push(property);
    }
    return __assign(__assign({}, (0, block_1.createBlock)(block)), { type: 'board', fields: {
            showDescription: (block === null || block === void 0 ? void 0 : block.fields.showDescription) || false,
            description: (block === null || block === void 0 ? void 0 : block.fields.description) || '',
            icon: (block === null || block === void 0 ? void 0 : block.fields.icon) || '',
            isTemplate: (block === null || block === void 0 ? void 0 : block.fields.isTemplate) || false,
            columnCalculations: (block === null || block === void 0 ? void 0 : block.fields.columnCalculations) || [],
            headerImage: (block === null || block === void 0 ? void 0 : block.fields.headerImage) || null,
            cardProperties: cardProperties
        } });
}
exports.createBoard = createBoard;
