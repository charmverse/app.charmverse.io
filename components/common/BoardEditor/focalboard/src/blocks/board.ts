import type { PageContent } from 'models';

import { IDType, Utils } from '../utils';

import type { Block } from './block';
import { createBlock } from './block';
import type { Card } from './card';

type PropertyType = 'text' | 'number' | 'select' | 'multiSelect' | 'date' | 'person' | 'file' | 'checkbox' | 'url' | 'email' | 'phone' | 'createdTime' | 'createdBy' | 'updatedTime' | 'updatedBy'

interface IPropertyOption {
    id: string;
    value: string;
    color: string;
}

// A template for card properties attached to a board
interface IPropertyTemplate {
    id: string;
    name: string;
    type: PropertyType;
    options: IPropertyOption[];
}

type BoardFields = {
    icon: string;
    description: PageContent;
    showDescription?: boolean;
    isTemplate?: boolean;
    cardProperties: IPropertyTemplate[];
    columnCalculations: Record<string, string>;
}

type Board = Block & {
    fields: BoardFields;
}

function createBoard ({ block, addDefaultProperty }: { block?: Block, addDefaultProperty?: boolean } | undefined = {}): Board {
  addDefaultProperty = addDefaultProperty ?? false;
  const cardProperties: IPropertyTemplate[] = block?.fields.cardProperties.map((o: IPropertyTemplate) => {
    return {
      id: o.id,
      name: o.name,
      type: o.type,
      options: o.options ? o.options.map((option) => ({ ...option })) : []
    };
  }) ?? [];

  const selectProperties = cardProperties.find((o) => o.type === 'select');

  if (!selectProperties && addDefaultProperty) {
    const property: IPropertyTemplate = {
      id: Utils.createGuid(IDType.BlockID),
      name: 'Status',
      type: 'select',
      options: [{
        color: 'propColorTeal',
        id: Utils.createGuid(IDType.BlockID),
        value: 'Completed'
      }, {
        color: 'propColorYellow',
        id: Utils.createGuid(IDType.BlockID),
        value: 'In progress'
      }, {
        color: 'propColorRed',
        id: Utils.createGuid(IDType.BlockID),
        value: 'Not started'
      }]
    };
    cardProperties.push(property);
  }

  return {
    ...createBlock(block),
    type: 'board',
    fields: {
      showDescription: block?.fields.showDescription || false,
      description: block?.fields.description || '',
      icon: block?.fields.icon || '',
      isTemplate: block?.fields.isTemplate || false,
      columnCalculations: block?.fields.columnCalculations || [],
      headerImage: block?.fields.headerImage || null,
      cardProperties
    }
  };
}

type BoardGroup = {
    option: IPropertyOption;
    cards: Card[];
}

export { createBoard };
export type { Board, PropertyType, IPropertyOption, IPropertyTemplate, BoardGroup };

