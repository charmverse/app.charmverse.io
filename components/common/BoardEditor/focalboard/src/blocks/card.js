"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
exports.__esModule = true;
exports.createCard = void 0;
var block_1 = require("./block");
/**
 * Returns a focalboard-ready card data stub
 */
function createCard(block) {
    var _a, _b, _c, _d, _e, _f;
    var contentOrder = [];
    var contentIds = (_b = (_a = block === null || block === void 0 ? void 0 : block.fields) === null || _a === void 0 ? void 0 : _a.contentOrder) === null || _b === void 0 ? void 0 : _b.filter(function (id) { return id !== null; });
    if ((contentIds === null || contentIds === void 0 ? void 0 : contentIds.length) > 0) {
        for (var _i = 0, contentIds_1 = contentIds; _i < contentIds_1.length; _i++) {
            var contentId = contentIds_1[_i];
            if (typeof contentId === 'string') {
                contentOrder.push(contentId);
            }
            else {
                contentOrder.push(contentId.slice());
            }
        }
    }
    return __assign(__assign({}, (0, block_1.createBlock)(block)), { type: 'card', fields: {
            icon: ((_c = block === null || block === void 0 ? void 0 : block.fields) === null || _c === void 0 ? void 0 : _c.icon) || '',
            properties: __assign({}, (((_d = block === null || block === void 0 ? void 0 : block.fields) === null || _d === void 0 ? void 0 : _d.properties) || {})),
            contentOrder: contentOrder,
            isTemplate: ((_e = block === null || block === void 0 ? void 0 : block.fields) === null || _e === void 0 ? void 0 : _e.isTemplate) || false,
            headerImage: ((_f = block === null || block === void 0 ? void 0 : block.fields) === null || _f === void 0 ? void 0 : _f.headerImage) || null
        } });
}
exports.createCard = createCard;
