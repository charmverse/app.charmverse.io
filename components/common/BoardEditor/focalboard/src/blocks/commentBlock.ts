import type { PageContent } from 'models';

import type { Block } from './block';
import { createBlock } from './block';

type CommentBlock = Block & {
  type: 'comment';
  fields: { content?: PageContent }; // note: the 'text' value is saved as block.title
}

function createCommentBlock (block?: Block): CommentBlock {
  return {
    ...createBlock(block),
    type: 'comment'
  };
}

export { createCommentBlock };
export type { CommentBlock };
