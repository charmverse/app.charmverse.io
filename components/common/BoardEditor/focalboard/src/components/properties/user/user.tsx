
import type { CSSObject } from '@emotion/serialize';
import Select from 'react-select';

import UserDisplay from 'components/common/UserDisplay';
import { useMembers } from 'hooks/useMembers';
import type { Member } from 'lib/members/interfaces';

import { getSelectBaseStyle } from '../../../theme';

type Props = {
  value: string;
  readOnly: boolean;
  onChange: (value: string) => void;
}

const selectStyles = {
  ...getSelectBaseStyle(),
  placeholder: (provided: CSSObject): CSSObject => ({
    ...provided,
    color: 'rgba(var(--center-channel-color-rgb), 0.4)'
  })
};

function UserProperty (props: Props): JSX.Element | null {
  const { members } = useMembers();
  const memberMap = members.reduce<Record<string, Member>>((acc, member) => {
    acc[member.id] = member;
    return acc;
  }, {});

  if (props.readOnly) {
    if (memberMap[props.value]) {
      return (
        <div className='UserProperty octo-propertyvalue'>
          <UserDisplay user={memberMap[props.value]} avatarSize='xSmall' fontSize='small' />
        </div>
      );
    }
    return null;
  }

  return (
    <Select
      options={members}
      isSearchable={true}
      isClearable={true}
      backspaceRemovesValue={true}
      className='UserProperty octo-propertyvalue'
      classNamePrefix='react-select'
      // eslint-disable-next-line react/no-unstable-nested-components
      formatOptionLabel={u => <UserDisplay user={u} avatarSize='small' fontSize='small' />}
      styles={selectStyles}
      placeholder='Empty'
      getOptionLabel={(o: Member) => o.username}
      getOptionValue={(a: Member) => a.id}
      value={memberMap[props.value] || null}
      onChange={(item, action) => {
        if (action.action === 'select-option') {
          props.onChange(item?.id || '');
        }
        else if (action.action === 'clear') {
          props.onChange('');
        }
      }}
    />
  );
}

export default UserProperty;
