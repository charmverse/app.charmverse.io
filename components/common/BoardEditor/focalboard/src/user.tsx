
export interface IUser {
    id: string;
    username: string;
    email: string;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    props: Record<string, any>;
    create_at: number;
    update_at: number;
    is_bot: boolean;
    wallet_address?: string;
}
