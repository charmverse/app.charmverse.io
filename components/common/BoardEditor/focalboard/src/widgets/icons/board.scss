.BoardIcon {
    stroke-width: 8px;
}
