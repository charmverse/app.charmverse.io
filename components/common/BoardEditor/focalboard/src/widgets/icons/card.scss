.CardIcon {
    stroke-width: 6px;
}
