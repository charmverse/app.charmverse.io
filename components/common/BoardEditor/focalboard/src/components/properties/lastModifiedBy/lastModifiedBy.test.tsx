// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import { Provider as ReduxProvider } from 'react-redux';

import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';

import { createCard } from '../../../blocks/card';
import { IUser } from '../../../user';

import { createBoard } from '../../../blocks/board';

import { createCommentBlock } from '../../../blocks/commentBlock';

import LastModifiedBy from './lastModifiedBy';

describe('components/properties/lastModifiedBy', () => {
  test('should match snapshot', () => {

    const mockStore = configureStore([]);
    const store = mockStore({
      users: {
        workspaceUsers: {
          'user-id-1': { username: 'username_1' } as IUser
        }
      }
    });

    const component = (
      <ReduxProvider store={store}>
        <LastModifiedBy
          updatedBy='user-id-1'
        />
      </ReduxProvider>
    );

    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });
});
