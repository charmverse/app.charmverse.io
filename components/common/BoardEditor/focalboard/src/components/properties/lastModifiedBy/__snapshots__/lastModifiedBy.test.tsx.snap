// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/properties/lastModifiedBy should match snapshot 1`] = `
<div>
  <div
    class="LastModifiedBy octo-propertyvalue readonly"
  >
    username_1
  </div>
</div>
`;
